app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -213,6 +216,18 @@ class SparkRestClient(sparkConf: SparkConf) {
       }
     }
   }
+
+  private def getStagesWithFailedTasks(attemptTarget: WebTarget): Seq[StageDataImpl] = {
+    val target = attemptTarget.path("stages/failedTasks")
```

Comment:
We have a separate API developed for "failed tasks". If we fetch data from the StageDatas, we will have to iterate over all the stages and filter out the stages having failed tasks, which is a costly operation, hence using this API to fetch failed tasks.

Reply:
If it's an API which is not available with the official version of Spark, then you need to add a comment about it. I am not sure if this PR should even be merged if this change is not going to work for most of the people.

I agree that iterating over all the stages could be a costly operation. But doing the same costly operation on Spark history server may be a lot more expensive because the history server is not only limited to Dr Elephant.

Reply:
Hi Shekhar 
As it turns out, the stages API does not return the task information. The only way to have failed task information is this or to call task API separately for each task. Hence, we will have to use this call only. 
You can comment out the new heuristics which require the new history server changes and you should be good to go.

Reply:
Hi Shekhar
I have made the calling of FailedTasksAPI configurable, the default value being false, now you don't have to worry about it as for the official version of spark, it won't call the API.

Reply:
Thanks, @skakker for making it configurable. However, I am still not in favor of merging this change because it's going to be useless for almost all the users unless they patch their SHS which is not a trivial exercise. 
@akshayrai @shankar37 What do you guys think about this issue?

Reply:
@shkhrgpt, we will not merge these PRs (which depend on the custom Spark HS) with the master as long as the changes are not reflected in the public spark release. 

We are planning to merge these to a separate branch for now.